src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Where.java

```diff
@@ -64,11 +65,12 @@ private void useAnyChildTag(Scope scope) {
     }
 
     /**
-     * Merge anyway if the root of WHERE clause be reached
+     * Merge anyway if the root of WHERE clause or {@link SQLNotExpr} be reached.
      */
-    private void mergeIfHaveTagAndIsRootOfWhere(Scope scope) {
-        if (!scope.getConditionTag(expr).isEmpty()
-                && expr.getParent() instanceof MySqlSelectQueryBlock) {
+    private void mergeIfHaveTagAndIsRootOfWhereOrNot(Scope scope) {
+        if ((!scope.getConditionTag(expr).isEmpty()
```

Comment:
this is personal opinion, but I think that the condition here is too long and hard to read. I'd recommend to refactor the method as following

```
private void mergeIfHaveTagAndIsRootOfWhereOrNot(Scope scope) {
    if (expr.getParent() instanceof SQLNotExpr) {
           mergeNestedField(scope);
           return;
    }

if (expr.getParent() instanceof MySqlSelectQueryBlock && !scope.getConditionTag(expr).isEmpty()) {
{   
    merge(..);
}
}

it makes clear that we're checking parent type and doing something here

Reply:
Done.